src/client.cpp

```diff
@@ -120,8 +120,8 @@ ADDON_STATUS ADDON_GetStatus()
   CLockObject lock(g_mutex);
 
   // Check that we're still connected
-  if (m_CurStatus == ADDON_STATUS_OK && !tvh->IsConnected())
-    m_CurStatus = ADDON_STATUS_LOST_CONNECTION;
+  if (m_CurStatus == ADDON_STATUS_OK && !tvh->IsSuspended() && !tvh->IsConnected())
+    return ADDON_STATUS_LOST_CONNECTION;
```

Comment:
@Jalle19 I think we need to fix some more addon status handling stuff in pvr.hts. Currently, there is a mixture of directly returning an ADDON_STATUS without setting m_CurStatus and setting m_CurStatus (before returning it). IMO this should be changed to consequently setting m_CurStatus on status change. IMO, we cannot get rid of m_CurrStatus, for instance because we need to implement ADDON_GetStatus properly. What do you think?

@FernetMenta pvr.hts already handles ADDON_STATUS_LOST_CONNECTION by itself (detect loss, automatically reconnect, ...). Only "problem" is to report loss/restore of connection back to the Kodi user. How should that work without this addon status? I'm not a friend of addon calling XBMC->QueueNotification as this is somehow "hardwired" and for instance gives no control whether (also) to write an activity log entry, for instance.